tests/test_application.py

```diff
@@ -50,7 +47,7 @@ class TestClientApplicationAcquireTokenSilentErrorBehaviors(unittest.TestCase):
 
     def setUp(self):
         self.authority_url = "https://login.microsoftonline.com/common"
-        self.authority = msal.authority.Authority(self.authority_url)
+        self.authority = msal.authority.Authority(self.authority_url, http_client= requests.Session())
```

Comment:
We briefly mentioned this before, that it would actually be valuable to implement a `MinimalHttpClient` (which will be what we had as `DefaultHttpClient` and `HttpResponse`), put it as `tests/http_client.py`, and then we use an instance of such minimal http client to perform all the `test_e2e.py` tests. We will also use a (new) instance of such a `MinimalHttpClient` in `test_authority.py`.